api/app/tests/apps/api/address/test_validation.py

```diff
@@ -7,17 +7,86 @@
+from signals.apps.api.v1.validation.address.bag import AddressValidation
+from signals.apps.api.v1.validation.address.base import (
     AddressValidationUnavailableException,
     NoResultsException
 )
+from signals.apps.api.v1.validation.address.pdok import PDOKAddressValidation
+
+
+class PDOKTestAddressValidation(SimpleTestCase):
+    PDOK_RESPONSE_JSON = "pdok_result.json"
+    pdok_response = None
+    address_dict = {
+        "openbare_ruimte": "Geuzenkade",
+        "huisnummer": 55,
+        "huisletter": "",
+        "huisnummer_toevoeging": "1",
+        "postcode": "1056KN",
+        "woonplaats": "Amsterdam"
+    }
+
+    def _get_mocked_response(self):
+        if self.pdok_response is None:
+            with open(join(dirname(abspath(__file__)), self.PDOK_RESPONSE_JSON)) as f:
+                self.pdok_response = load(f)
+        return self.pdok_response
+
+    def test_no_results(self):
+        address_validation = PDOKAddressValidation()
+        address_validation._search = MagicMock(return_value=[])
+
+        self.assertRaises(NoResultsException, address_validation.validate_address, self.address_dict)
+
+        address_validation._search.assert_called_with(self.address_dict)
+
+    def test_address_found(self):
+        address_validation = PDOKAddressValidation()
+
+        result = self._get_mocked_response()["response"]["docs"][0]
+
+        address_validation._search = MagicMock(return_value=[result])
+        address_validation._search_result_to_address = MagicMock()
+
+        address_validation.validate_address(self.address_dict)
+
+        address_validation._search.assert_called_with(self.address_dict)
+        address_validation._search_result_to_address.assert_called_with(result)
+
+    def test_search_unavailable(self):
+        address_validation = PDOKAddressValidation()
+
+        with Mocker() as m:
+            m.get(address_validation.address_validation_url, status_code=400)
+            self.assertRaises(AddressValidationUnavailableException, address_validation._search, self.address_dict)
+
+    def test_search_connection_error(self):
+        address_validation = PDOKAddressValidation()
+
+        with Mocker() as m:
+            m.get(address_validation.address_validation_url, exc=ConnectionError)
+            self.assertRaises(AddressValidationUnavailableException, address_validation._search, self.address_dict)
+
+    def test_search_successful_request(self):
+        address_validation = PDOKAddressValidation()
+
+        result = self._get_mocked_response()
+        expected = result["response"]["docs"]
+
+        with Mocker() as m:
+            m.get(address_validation.address_validation_url, text=dumps(result))
+            self.assertEqual(address_validation._search(self.address_dict), expected)
 
 
 class TestAddressValidation(SimpleTestCase):
     ATLAS_RESPONSE_JSON = "atlas_result.json"
     atlas_response = None
     address = "Weesperstraat 113"
+    address_dict = {
```

Comment:
This class should probably be called: `TestBAGAddressValidation`.

Reply:
Changed